official/vision/beta/projects/yolo/modeling/layers/nn_blocks_test.py

```diff
@@ -281,5 +283,102 @@ def test_gradient_pass_though(self, filters, width, height, downsample):
 
     self.assertNotIn(None, grad)
 
+
+class DarkSppTest(tf.test.TestCase, parameterized.TestCase):
+
+  @parameterized.named_parameters(("RouteProcessSpp", 224, 224, 3, [5, 9, 13]),
+                                  ("test1", 300, 300, 10, [2, 3, 4, 5]),
+                                  ("test2", 256, 256, 5, [10]))
+  def test_pass_through(self, width, height, channels, sizes):
+    x = tf.keras.Input(shape=(width, height, channels))
+    test_layer = nn_blocks.SPP(sizes=sizes)
+    outx = test_layer(x)
+    self.assertAllEqual(outx.shape.as_list(),
+                        [None, width, height, channels * (len(sizes) + 1)])
+    return
+
+  @parameterized.named_parameters(("RouteProcessSpp", 224, 224, 3, [5, 9, 13]),
+                                  ("test1", 300, 300, 10, [2, 3, 4, 5]),
+                                  ("test2", 256, 256, 5, [10]))
+  def test_gradient_pass_though(self, width, height, channels, sizes):
+    loss = tf.keras.losses.MeanSquaredError()
+    optimizer = tf.keras.optimizers.SGD()
+    test_layer = nn_blocks.SPP(sizes=sizes)
+
+    init = tf.random_normal_initializer()
+    x = tf.Variable(
+        initial_value=init(
+            shape=(1, width, height, channels), dtype=tf.float32))
+    y = tf.Variable(
+        initial_value=init(
+            shape=(1, width, height, channels * (len(sizes) + 1)),
+            dtype=tf.float32))
+
+    with tf.GradientTape() as tape:
+      x_hat = test_layer(x)
+      grad_loss = loss(x_hat, y)
+    grad = tape.gradient(grad_loss, test_layer.trainable_variables)
+    optimizer.apply_gradients(zip(grad, test_layer.trainable_variables))
+
+    self.assertNotIn(None, grad)
+    return
+
+
+class DarkRouteProcessTest(tf.test.TestCase, parameterized.TestCase):
+
+  @parameterized.named_parameters(
+      ("test1", 224, 224, 64, 7, False), ("test2", 223, 223, 32, 3, False),
+      ("tiny", 223, 223, 16, 1, False), ("spp", 224, 224, 64, 7, False))
+  def test_pass_through(self, width, height, filters, repetitions, spp):
+    x = tf.keras.Input(shape=(width, height, filters))
+    test_layer = nn_blocks.DarkRouteProcess(
+        filters=filters, repetitions=repetitions, insert_spp=spp)
+    outx = test_layer(x)
+    self.assertEqual(len(outx), 2, msg="len(outx) != 2")
+    if repetitions == 1:
```

Comment:
DRY, you have some of this code in both test cases. Consider a class helper method.